### PR TITLE
update ocaml-lsp-server

### PR DESCRIPTION
**Pull request by tatchi**
This one update the `ocaml-lsp-server` dependencies (just removed the specific commit hash). I had to upgrade `dune` and `ocaml` as well to satisfy dependencies constraints.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -43,14 +43,14 @@
     "refmterr": "^3.3.0"
   },
   "devDependencies": {
-    "ocaml": "4.8.x",
-    "@opam/dune": "2.0.1",
+    "ocaml": "4.9.x",
```
Comment (tatchi) [inline package.json:46]: Should I update the `ocaml` dependencies as well? i.e: from `4.6.10 - 4.8` to `4.6.10 - 4.9` ?

Comment (prometheansacrifice) [reply package.json:46]: Yes, let's do that.

Comment (tatchi) [reply package.json:46]: done

**Comment by prometheansacrifice** [comment]
Thanks @tatchi This was very nicely done :)

**Pull request merged**